src/mqtt/new_mqtt.c

```diff
@@ -12,6 +12,8 @@
 #include "../driver/drv_tuyaMCU.h"
 #include "../ota/ota.h"
 
+#define BUILD_AND_VERSION_FOR_MQTT PLATFORM_MCU_NAME " " __DATE__ " " __TIME__ " v " USER_SW_VER
```

Comment:
#define BUILD_AND_VERSION_FOR_MQTT "OpenBeken v" USER_SW_VER " hw: " PLATFORM_MCU_NAME " mf: " __MANUFACTURER_STRING__ " build: " __DATE__ " " __TIME__

Something like this maybe? Still not perfect; I'm trying to order it both logically, but also where data is likely to be parsed, and I reckon the date/time is the least likely thing anyone will want to parse out.